docs/enhancements/ai-on-zero-worker-hypershift.md

```diff
@@ -0,0 +1,118 @@
+---
+title: ai-on-zero-worker-hypershift
+authors:
+  - "@slaviered"
+creation-date: 2022-08-28
+last-updated: yyyy-mm-dd
+---
+#  Installing the Infrastructure Operator on zero-node HyperShift (DRAFT)
+
+# Summary
+To support large numbers of HyperShift hosted control planes on a single management cluster, an architecture of nested control planes has been proposed.  The management cluster, which we refer to as L0, runs hypershift hosted control planes (L1). Spoke clusters are added to L1 control planes. The challenge is that we need to run the assisted-installer in L1 for adding workers to spoke clusters, but L1 doesn't have any nodes and therefore cannot run the installer directly.
+
+The architecture for multiple hubs is described in the following diagram
+<img src="https://github.com/xiangjingli/hyper-acm-deploy/blob/main/docs/images/acm-hosted-cluster-diagram.svg" width="30%" height="30%"/>
+
+The hosted control plane (L1) is responsible for managing the CRs and CRDs, where as the actual pods are located in dedicated namespaces and managed by the hosting management cluster (L0).
+
+The deployment and cluster provisioning on such a setup is described in this [POC](https://github.com/javipolo/openshift-assisted-installer-tests/tree/main/ai-in-zero-worker-hypershift). For convinience, the architecure of the deployment is also pasted ![here:](../ai-on-hypershift.png).
+
+# Motivation
+## Goals
+- Automate the deployment of assisted service on zero-nodeworker hypershift clusters using the assisted infrastructure operator
+- Support Infrastructure Operatorai installation on multiple namespaces.
+- Isolation: Each instance of assisted service controls the spoke cluster creation on its corresponding hypershift-based hub cluster
+- Backward Compatiability: Support automatic update of the infrastructure operator and preserve the current behavior on non-hypershift installations.
+
+# Proposal
+## User Story
+1. The user installs the assisted insfrastructure operator
```

Comment:
These are not really user stories, right? This is more of an example flow.

For a user story, I'd expect something more like:


`As an administrator, I can install assisted service into several namespaces in order for each to watch resources in a hosted control plane`

Then maybe something about how a user should expect assisted service to behave when installing a cluster.

Reply:
To be clear though, I think this list of events is useful but should be put somewhere else. Maybe in the implementation details sections.

Reply:
I changed the title to reflect that we are listing the installation steps from the user's perspective. It is a bit out of place in the Implementation Details section so let's keep it in a separate section.